File: app/src/main/java/com/risalfajar/kamusoxford/Entity/Word.java
package com.risalfajar.kamusoxford.Entity;

public class Word {
    String word;
    String origin;
    LexicalEntry[] lexicalEntries;
}

class LexicalEntry{
    Entry[] entries;
    String lexicalCategory;
    Pronouncitation[] pronouncitations;
    String text;
}

class Entry{
    String origin;
    Sense[] senses;
}

class Sense{
    String[] definitions;
    String[] examples;
    String[] shortDefinitions;
    Subsense[] subsenses;
}

class Subsense{
    String[] definitions;
    String[] examples;
}

class Pronouncitation{
    String audioURL;
    String[] dialects;
}

File: app/src/main/java/com/risalfajar/kamusoxford/service/entity/Word.kt
package com.risalfajar.kamusoxford.service.entity

import java.util.*

data class Word(var id: String) {
    var origin: String? = null
    var lexicalEntries: ArrayList<LexicalEntry>? = null
}

class LexicalEntry {
    var entries: ArrayList<Entry>? = null
    var lexicalCategory: String? = null
    var pronouncitations: ArrayList<Pronouncitation>? = null
    var text: String? = null
}

class Entry {
    var origin: String? = null
    var senses: ArrayList<Sense>? = null
}

class Sense {
    var definitions: Array<String>? = null
    var examples: Array<String>? = null
    var shortDefinitions: Array<String>? = null
    var subsenses: ArrayList<Subsense>? = null
}

class Subsense {
    var definitions: Array<String>? = null
    var examples: Array<String>? = null
}